syncd/syncd_counters.cpp

```diff
@@ -0,0 +1,299 @@
+    ENTRY(SAI_PORT_STAT_ETHER_STATS_CRC_ALIGN_ERRORS);
+    ENTRY(SAI_PORT_STAT_ETHER_STATS_TX_NO_ERRORS);
+    ENTRY(SAI_PORT_STAT_ETHER_STATS_RX_NO_ERRORS);
+    ENTRY(SAI_PORT_STAT_IP_IN_RECEIVES);
+    ENTRY(SAI_PORT_STAT_IP_IN_OCTETS);
+    ENTRY(SAI_PORT_STAT_IP_IN_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IP_IN_NON_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IP_IN_DISCARDS);
+    ENTRY(SAI_PORT_STAT_IP_OUT_OCTETS);
+    ENTRY(SAI_PORT_STAT_IP_OUT_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IP_OUT_NON_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IP_OUT_DISCARDS);
+    ENTRY(SAI_PORT_STAT_IPV6_IN_RECEIVES);
+    ENTRY(SAI_PORT_STAT_IPV6_IN_OCTETS);
+    ENTRY(SAI_PORT_STAT_IPV6_IN_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IPV6_IN_NON_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IPV6_IN_MCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IPV6_IN_DISCARDS);
+    ENTRY(SAI_PORT_STAT_IPV6_OUT_OCTETS);
+    ENTRY(SAI_PORT_STAT_IPV6_OUT_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IPV6_OUT_NON_UCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IPV6_OUT_MCAST_PKTS);
+    ENTRY(SAI_PORT_STAT_IPV6_OUT_DISCARDS);
+    ENTRY(SAI_PORT_STAT_GREEN_DISCARD_DROPPED_PACKETS);
+    ENTRY(SAI_PORT_STAT_GREEN_DISCARD_DROPPED_BYTES);
+    ENTRY(SAI_PORT_STAT_YELLOW_DISCARD_DROPPED_PACKETS);
+    ENTRY(SAI_PORT_STAT_YELLOW_DISCARD_DROPPED_BYTES);
+    ENTRY(SAI_PORT_STAT_RED_DISCARD_DROPPED_PACKETS);
+    ENTRY(SAI_PORT_STAT_RED_DISCARD_DROPPED_BYTES);
+    ENTRY(SAI_PORT_STAT_DISCARD_DROPPED_PACKETS);
+    ENTRY(SAI_PORT_STAT_DISCARD_DROPPED_BYTES);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_64_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_65_TO_127_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_128_TO_255_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_256_TO_511_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_512_TO_1023_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_1024_TO_1518_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_1519_TO_2047_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_2048_TO_4095_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_4096_TO_9216_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_IN_PKTS_9217_TO_16383_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_64_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_65_TO_127_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_128_TO_255_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_256_TO_511_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_512_TO_1023_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_1024_TO_1518_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_1519_TO_2047_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_2048_TO_4095_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_4096_TO_9216_OCTETS);
+    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_9217_TO_16383_OCTETS);
+
+    return v;
+}
+
+std::string getCounterName(sai_port_stat_counter_t counter)
+{
+    if (counter < g_counterNameVector.size())
+        return g_counterNameVector[counter];
+
+    // TODO exit here ?
+    return std::to_string(counter);
+}
+
+sai_port_stat_counter_t collectCountersIds[] = {
```

Comment:
Is there a ‘get sai capabilities’ method or some way a SAI implementation can express which attributes it supports?


From: lguohan [mailto:notifications@github.com]
Sent: Friday, April 29, 2016 6:55 AM
To: Azure/sonic-sairedis <sonic-sairedis@noreply.github.com>
Subject: Re: [Azure/sonic-sairedis] Add port counters collector thread (#12)


In syncd/syncd_counters.cpp<https://github.com/Azure/sonic-sairedis/pull/12#discussion_r61580858>:

> +    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_4096_TO_9216_OCTETS);

> +    ENTRY(SAI_PORT_STAT_ETHER_OUT_PKTS_9217_TO_16383_OCTETS);

> +

> +    return v;

> +}

> +

> +std::string getCounterName(sai_port_stat_counter_t counter)

> +{

> +    if (counter < g_counterNameVector.size())

> +        return g_counterNameVector[counter];

> +

> +    // TODO exit here ?

> +    return std::to_string(counter);

> +}

> +

> +sai_port_stat_counter_t collectCountersIds[] = {

not sure if mlnx supports all these counters. I remember mlnx might not support some of those counters, for example IF_OUT_QLEN.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/Azure/sonic-sairedis/pull/12/files/0fd61e032c668e1fe57645defdcf6d8060a9ed61#r61580858>